src/dfx/src/commands/deploy.rs

```diff
@@ -212,15 +211,34 @@ fn display_urls(env: &dyn Environment) -> DfxResult {
                 // If the canister is an assets canister or has a frontend section, we can display a frontend url.
                 let is_assets = canister_info.is_assets() || canister_config.frontend.is_some();
 
+                let provider = Url::parse(&network.providers[0]).with_context(|| {
+                    format!(
+                        "Failed to parse url for network provider {}.",
+                        &network.providers[0]
+                    )
+                })?;
                 if is_assets {
-                    let url = construct_frontend_url(network, &canister_id)?;
+                    let url = format_frontend_url(&provider, &canister_id.to_string());
                     frontend_urls.insert(canister_name, url);
                 }
 
                 if !canister_info.is_assets() {
-                    let url = construct_ui_canister_url(network, &canister_id, ui_canister_id)?;
-                    if let Some(ui_canister_url) = url {
-                        candid_urls.insert(canister_name, ui_canister_url);
+                    let is_local = env.get_network_descriptor().name == "local";
```

Comment:
What's the rationale in changing this logic?  This used to check `network.is_ic`, which matches the [condition for creating](https://github.com/dfinity/sdk/blob/master/src/dfx/src/lib/operations/canister/install_canister.rs#L58) the candid UI canister.

Also the name of `format_ui_canister_url_ic` is a hint: "do this on IC", not "do this unless the network name is 'local'"